pkgs/applications/misc/syncthingtray/default.nix

```diff
@@ -0,0 +1,53 @@
+{ mkDerivation, stdenv, lib, fetchFromGitHub
+, qtbase, extra-cmake-modules, cpp-utilities, qtutilities
+, cmake, kio, plasma-framework, qttools
+, webviewProvider ? null
+, jsProvider ? null
+, enableKioPlugin ? false
+, enablePlasmoid  ? false
+, systemdSupport ? true }:
+
+mkDerivation rec {
+  version = "0.9.1";
+  pname = "syncthingtray";
+
+  src = fetchFromGitHub {
+    owner  = "Martchus";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "0ijwpwlwwbfh9fdfbwz6dgi6hpmaav2jm56mzxm6as50iwnb59fx";
+  };
+
+  buildInputs = [ qtbase cpp-utilities qtutilities webviewProvider jsProvider ] 
+    ++ lib.optional enableKioPlugin kio
+    ++ lib.optional enablePlasmoid extra-cmake-modules
+    ++ lib.optional enablePlasmoid plasma-framework
```

Comment:
I coldn't figure out the differences from [the docs](https://nixos.org/nixpkgs/manual/#function-library-lib.lists.optionals).. with `lib.optionals` just used instead of `lib.optional`, I get:

```
error: value is a string while a list was expected, at /var/src/nixpkgs/pkgs/applications/misc/syncthingtray/default.nix:32:8
```

Reply:
`optionals` takes a list and returns a list, whilel `optional` takes a single value and returns a list. If you still have issues please paste the troublesome line.

Reply:
Got it, thanks.